### DISCUSSION TITLE
pinout

### DISCUSSION THREAD

**Discussion by Bill-Hopper520** [grblHAL/RP2040#52] [Q&A]
Before I begin with my issues, I would like to make a shout out to those involved in making this HAL, I think it is phenomenal! I love the pico and was so excited to utilize it to upgrade my machine. The current setup that I have uses TB6600 stepper drivers for all axis and auxiliary hardware(locks for access doors/valves for suction and coolant). I have been able to integrate all auxiliary HW using a second mcu that interprets signals from the pico to control these. Thus my problem: I spent some serious time figuring out how to eliminate an issue I found prior to beginning the install, but  now I am having trouble determining the pins for the "Step Pulse" in the generic map. If anybody could provide insight, I would really appreciate it.

**Comment by terjeio** [comment]
The generic map makes use of the PIO peripheral to generate step pulses, this has a mode where it can efficiently use pins if they are consecutive.

`#define STEP_PINS_BASE          2         // N_AXIS number of consecutive pins are used by PIO`
This means that step output for the X-axis is on GP2, Y-axis on GP3 and Z-axis on GP4.

When revisiting the generic map I see now that pins overlap if compiling for a 4-axis machine. I'll remove that capability and create a new generic map that can support 4 axes.

**Reply by lateforwork1000** [reply]
I don't really understand this is there something i can use to reference to get the pin out?

**Comment by Bill-Hopper520** [comment]
Thank you for the clarification, the overlap was confusing me
